tests/v2/validation/rbac/rbac_namespace_move_test.go

```diff
@@ -0,0 +1,220 @@
+	moveNameSpaceFromAProjectRoleToAProject(rb, rb.standardUserProjectOwner, roleProjectOwner, rb.standardUserProjectReadOnly, roleProjectReadOnly, rb.steveAdminClient, true)
+}
+
+func (rb *NameSpaceMoveTestSuite) validateAdminCanMoveANamespaceFromProjectMember() {
+	moveNameSpaceFromAProjectRoleToAProject(rb, rb.standardUserProjectReadOnly, roleProjectReadOnly, rb.standardUserProjectOwner, roleProjectOwner, rb.steveAdminClient, true)
+}
+
+func (rb *NameSpaceMoveTestSuite) validateAdminCanMoveANamespaceFromPMToReadOnlyProject() {
+	moveNameSpaceFromAProjectRoleToAProject(rb, rb.standardUserProjectMember, roleProjectMember, rb.standardUserProjectReadOnly, roleProjectReadOnly, rb.steveAdminClient, true)
+}
+
+func (rb *NameSpaceMoveTestSuite) validateAdminWithClusterOwnerCanMoveANamespaceFromProjectReadOnlyPermission() {
+	moveNameSpaceFromAProjectRoleToAProject(rb, rb.standardUserProjectOwner, roleOwner, rb.standardUserProjectReadOnly, roleProjectReadOnly, rb.steveAdminClient, true)
+}
+
+func (rb *NameSpaceMoveTestSuite) validateProjectOwnerCannotMoveANamespaceToReadOnlyProject() {
+	steveStdUserclient, err := rb.client.Steve.ProxyDownstream(rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	rb.steveNonAdminClient = steveStdUserclient
+
+	moveNameSpaceFromAProjectRoleToAProject(rb, rb.standardUserProjectOwner, roleProjectOwner, rb.standardUserProjectReadOnly, roleProjectReadOnly, rb.steveNonAdminClient, false)
+}
+
+func (rb *NameSpaceMoveTestSuite) validateProjectMemberCannotMoveANamespaceToReadOnlyProject() {
+	steveStdUserclient, err := rb.client.Steve.ProxyDownstream(rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	rb.steveNonAdminClient = steveStdUserclient
+
+	moveNameSpaceFromAProjectRoleToAProject(rb, rb.standardUserProjectMember, roleProjectMember, rb.standardUserProjectReadOnly, roleProjectReadOnly, rb.steveNonAdminClient, false)
+}
+
+func (rb *NameSpaceMoveTestSuite) ValidateStandardUserProjectReadOnlyCannotMoveANamespaceWithReadOnlyProjectPermission() {
+	steveStdUserclient, err := rb.client.Steve.ProxyDownstream(rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	rb.steveNonAdminClient = steveStdUserclient
+
+	moveNameSpaceFromAProjectRoleToAProject(rb, rb.standardUserProjectReadOnly, roleProjectReadOnly, rb.standardUserProjectReadOnly, roleProjectReadOnly, rb.steveNonAdminClient, false)
+}
+
+func (rb *NameSpaceMoveTestSuite) validateProjectReadOnlyCannotMoveANamespaceToReadOnlyProject() {
+	steveStdUserclient, err := rb.client.Steve.ProxyDownstream(rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	rb.steveNonAdminClient = steveStdUserclient
+
+	moveNameSpaceFromAProjectRoleToAProject(rb, rb.standardUserProjectOwner, roleOwner, rb.standardUserProjectReadOnly, roleProjectReadOnly, rb.steveNonAdminClient, false)
+}
+
+func (rb *NameSpaceMoveTestSuite) TestMoveNamespaceToNewProjectTestSuite() {
+
+	rb.Run("Create three projects with different roles", func() {
+		rb.createProjects()
```

Comment:
Why is this a test case? This sounds like a setup, since it is for the whole life cycle of the test.